awsmp/_driver.py

```diff
@@ -323,19 +319,37 @@ def _get_pricing_diff(product_id: str, changeset: List[ChangeSetType]) -> Tuple[
     diffs_hourly = _build_pricing_diff(existing_hourly, local_hourly)
     diffs_annual = _build_pricing_diff(existing_annual, local_annual)
 
-    if existing_listing_status != "Draft":
-        # Only staging listings are able to change pricing type
-        # e.g. hourly -> hourly + annual, hourly + annual -> hourly
-        pricing_type = (local_annual and not existing_annual) or (existing_annual and not local_annual)
+    if existing_listing_status == "Restricted":
+        # restricted instances do not support updating instance types
+        error_message = "Restricted listings may not have instance types updated."
+        raise AmiPriceChangeError(error_message)
 
-        def all_zero_to_paid(diffs):
-            # check if pricing request from free (0.0) to non-zero prices
-            return bool(diffs) and all(
-                float(item["Original Price"]) == 0.0 and float(item["New Price"]) != 0.0 for item in diffs
-            )
+    def any_zero_to_paid(diffs):
+        # check if pricing request from free (0.0) to non-zero prices
+        return bool(diffs) and any(
+            float(item["Original Price"]) == 0.0 and float(item["New Price"]) != 0.0 for item in diffs
+        )
 
-        if pricing_type or all_zero_to_paid(diffs_hourly) or all_zero_to_paid(diffs_annual):
-            raise AmiPriceChangeError
+    instance_configuration_changed = any(
+        [local_annual and not existing_annual, existing_annual and not local_annual, diffs_annual, diffs_hourly]
+    )
+
+    if (any_zero_to_paid(diffs_hourly) or any_zero_to_paid(diffs_annual)) and not allow_price_update:
+        error_msg = f"""Free product was attempted to be converted to paid product.
+            Please check the pricing files or set the price flag.\n
+            Price change details:\n
+            Local pricing updates: {local_annual}\Existing pricing in local: {existing_annual}\n"
+            """
+        logger.error(error_msg)
+        raise AmiPriceChangeError(error_msg)
+    elif instance_configuration_changed and not allow_price_update:
+        error_message = f"""There are pricing changes in either hourly or annual prices.
+        Please check the pricing files or allow price change.
+        Price change details:\n
+        Local pricing updates: {local_annual}\Existing pricing in local: {existing_annual}\n"
+        """
+        logger.error(error_message)
+        raise AmiPriceChangeError(error_message)
```

Comment:
If I'm reading correctly instance_configuration_changed will flag on the the addition or removal of annual prices (e.g. switching from hourly+annual to just hourly or vice versa). Those changes must be done via support ticket.

Breaking those cases out into a separate elif and raising a different error with similar messaging to the old AmiPriceChangeError should be sufficient (`AmiPriceModelChangeError` perhaps?)

Reply:
:done: